include/nanoeigenpy/decompositions/EigenSolver.hpp

```diff
@@ -0,0 +1,81 @@
+/// Copyright 2025 INRIA
+#pragma once
+#include "nanoeigenpy/eigen-base.hpp"
+// #include "nanoeigenpy/eigen-to-python.hpp"
+#include <Eigen/Core>
+#include <Eigen/Eigenvalues>
+#include <optional>
+
+namespace nanoeigenpy {
+namespace nb = nanobind;
+
+template <typename MatrixType>
+Eigen::EigenSolver<MatrixType> &compute_proxy(
+    Eigen::EigenSolver<MatrixType> &c,
+    const Eigen::EigenBase<MatrixType> &matrix) {
+  return c.compute(matrix);
+}
```

Comment:
This is 2025, you can use a lambda instead, i.e. `[](Solver &s, const MatrixType &mat) -> auto& { return s.compute(mat); } `